flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/nodes/dataset/DataSetValues.scala

```diff
@@ -67,7 +67,7 @@ class DataSetValues(
   }
 
   override def translateToPlan(
-      tableEnv: BatchTableEnvironment,
+      tableEnv: BatchTableEnvImpl,
```

Comment:
Do we still need to pass the implementation through the entire planner? Could you check what is really required in the planner? Maybe we can introduce some `PlannerContext` class instead?

Reply:
Totally agree with you. Most of the places only use `tableEnv.getConfig`. However, `tableEnv.execEnv` is used in `DataSetValues`, `DataStreamValues` and `StreamTableSourceScan`.  Also `tableEnv.getRelBuilder` is used in `StreamTableSourceScan`. Maybe we can simplify this a bit or just simply put them into the `PlannerContext`. 

Btw, how about fixing this in another PR. We may need some more discussions. And let this PR focus on the tableEnv interface. What do you think?

Reply:
Totally agree. I thought we are only using `getConfig()`. We can postpone any bigger change here. This code will be dropped anyway soon.